docs/technical_reference/costing/automate_detailed_unit_costing_rst_file.py

```diff
@@ -0,0 +1,105 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2024, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+__author__ = "Adam Atia"
+
+import pandas as pd
+
+df = pd.read_csv("WT_unit_costing_for_docs.csv")
+
+unit_title_list = [i.title() for i in df["name"]]
+
+unit_name_list = df["unit"]
+
+title_exceptions = {
+    "cstr_injection": "Completely Stirred Tank Reactor w/Injection Stream",
+    "cstr": "Completely Stirred Tank Reactor",
+    "dewatering": "Dewatering Unit",
+    "electroNP": "ElectroN-P",
+    "gac": "Granular Activated Carbon",
+    "heater_chiller": "Heater/Chiller",
+    "uv_aop": "UV with Advanced Oxidation Processes",
+}
+
+
+if __name__ == "__main__":
+
+    # Create index file for all unit model costing docs
+    with open("detailed_unit_model_costing.rst", "w") as f:
+        f.write("Detailed Unit Model Costing\n")
+        f.write("=" * len("Detailed Unit Model Costing"))
+        f.write("\n")
+        f.write(
+            "Default costing methods are provided for the unit models listed below. However, users should supply their own cost relationship, if possible, instead of relying completely on the defaults.\n\n"
+        )
+        f.write(".. toctree::\n")
+        f.write("   :maxdepth: 1\n\n")
+
+    for i, u in enumerate(unit_title_list):
+
+        # list = [
+        #     f"This unit model is formulated as a **{model_type_list[i]}** model form.",
+        #     f"See documentation for :ref:`{model_type_list[i]} Helper Methods<{model_type_ref_list[i]}>`.",
+        #     f"Electricity consumption is calculated using the **{elect_func_list[i]}** helper function.",
+        #     None,  # created on the fly now
+        #     f"   pair: watertap.unit_models.zero_order.{unit_name_list[i]};{unit_name_list[i]}",
+        #     f".. currentmodule:: watertap.unit_models.zero_order.{unit_name_list[i]}",
+        #     f".. automodule:: watertap.unit_models.zero_order.{unit_name_list[i]}",
+        # ]
+
+        # append unit doc to index
+        with open("detailed_unit_model_costing.rst", "a") as f:
+            f.write(f"   {unit_name_list[i]}\n")
+
+        with open(f"{unit_name_list[i]}.rst", "w", encoding="utf-8") as f:
```

Comment:
So we can reuse this when more models are added, I think this should check for the existence of f"{unit_name_list[i]}.rst" before opening this file.

Reply:
As in- "do not overwrite the file if it exists already"?

Reply:
Or simply "only open and write to the file if it has been created already"?

Reply:
> As in- "do not overwrite the file if it exists already"?

This one -- I assume these will be filled-out in subsequent PRs, and we don't want to overwrite files which already exist.

Reply:
I was originally contemplating whether to automatically fill the tables with model params/vars as was done previously for the ZO model costing documentation. However, I am thinking we might as well just let each assignee manually fill out the relevant sections without needing to run this script (at least for the time being.

I can add a boolean option to enable/disable overwriting here, but unsure how necessary it would be if we turn the guidance into--don't run the script--just manually edit your rst file as needed.

Reply:
Actually, we'll want you're suggestion for later on, when new unit model methods get added. So I will try to add it in this PR.